examples/threading-lock/src/main.rs

```diff
@@ -0,0 +1,47 @@
+#![no_main]
+#![no_std]
+#![feature(type_alias_impl_trait)]
+#![feature(used_with_arg)]
+
+use embassy_time::{Duration, Timer};
+use riot_rs::{debug::log::*, thread::lock::Lock, thread::ThreadId};
+
+static LOCK: Lock = Lock::new();
+
+#[riot_rs::task(autostart)]
+async fn task_with_timer() {
+    info!("Task waits for lock");
+    LOCK.acquire();
```

Comment:
This lock would put the current thread to sleep. So when called from a thread executor, this blocks the whole executor. if called in an interrupt executor, it violates the huge `**NOTE**` in the `acquire()`docs...

Reply:
Good point. The idea was to give the other threads a chance to run, while holding the lock, but I solved it differently now.